utilities/infra.py

```diff
@@ -1536,3 +1537,17 @@ def verify_image_info(image_url, generated_pulled_secret, nodes_cpu_architecture
             f"--registry-config={generated_pulled_secret} --filter-by-os={nodes_cpu_architecture}"
         ),
     )
+
+
+def assert_os_version_mismatch_in_vm(vm: utilities.virt.VirtualMachineForTests, expected_os: str) -> None:
```

Comment:
since this is used other within infra/golden_images should we consider moving to local utils file to this folder?
rest lgtm

Reply:
to which location? I think we plan to remove the templates test in the future so for now it is a good location

Reply:
Geetika is right, this function is only used in infrastructure/golden_images, please move it to `tests/infrastructure/golden_images/utils.py`